dissect/target/plugins/apps/remoteaccess/anydesk.py

```diff
@@ -1,91 +1,109 @@
 
     def check_compatible(self) -> None:
-        if not (len(self.logfiles)):
-            raise UnsupportedPluginError("No Anydesk logs found")
+        if not self.trace_files:
+            raise UnsupportedPluginError("No Anydesk trace files found on target")
 
-    @export(record=RemoteAccessRecord)
-    def logs(self):
-        """Return the content of the AnyDesk logs.
+    @export(record=RemoteAccessLogRecord)
+    def logs(self) -> Iterator[RemoteAccessLogRecord]:
+        """Parse AnyDesk trace files.
 
         AnyDesk is a remote desktop application and can be used by adversaries to get (persistent) access to a machine.
-        Log files (.trace files) are retrieved from various location based on OS and client type.
+        Log files (.trace files) can be stored on various locations, based on target OS and client type.
+        Timestamps in trace files do not carry a time zone designator (TZD) but are in fact UTC.
 
         References:
             - https://www.inversecos.com/2021/02/forensic-analysis-of-anydesk-logs.html
             - https://support.anydesk.com/knowledge/trace-files#trace-file-locations
         """
-        for logfile, user in self.logfiles:
-            logfile = self.target.fs.path(logfile)
-
-            for line in logfile.open("rt"):
+        for trace_file, user in self.trace_files:
+            for line in trace_file.open("rt", errors="backslashreplace"):
                 line = line.strip()
 
-                # Skip empty lines
-                if not line:
+                if not line or "* * * * * * * * * * * * * *" in line:
                     continue
 
-                if "* * * * * * * * * * * * * *" in line:
-                    continue
+                try:
+                    level, ts_date, ts_time, message = line.split(" ", 3)
 
-                level, ts_day, ts_time, description = line.split(" ", 3)
-                description = f"{level} {description}"
-                ts_time = ts_time.split(".")[0]
+                    timestamp = datetime.strptime(f"{ts_date} {ts_time}", "%Y-%m-%d %H:%M:%S.%f").replace(tzinfo=timezone.utc)
+                    message = re.sub(r"\s\s+", " ", f"{level} {message}")
 
-                timestamp = datetime.strptime(f"{ts_day} {ts_time}", "%Y-%m-%d %H:%M:%S")
+                    yield self.RemoteAccessLogRecord(
+                        ts=timestamp,
+                        message=message,
+                        source=trace_file,
+                        _target=self.target,
+                        _user=user,
+                    )
 
-                yield RemoteAccessRecord(
-                    ts=timestamp,
-                    tool="anydesk",
-                    logfile=str(logfile),
-                    description=description,
-                    _target=self.target,
-                    _user=user,
-                )
+                except (IndexError, ValueError) as e:
```

Comment:
I'm probably overlooking but I don't think an `IndexError` should be possible anymore?